external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -118,9 +123,15 @@ fn extract_macros(context: &mut Context, modules: &UniqueMap<ModuleIdent, N::Mod
 
 fn modules(
     context: &mut Context,
-    modules: UniqueMap<ModuleIdent, N::ModuleDefinition>,
+    mut modules: UniqueMap<ModuleIdent, N::ModuleDefinition>,
 ) -> UniqueMap<ModuleIdent, T::ModuleDefinition> {
     let mut all_new_friends = BTreeMap::new();
+    // We validate the syntax methods first so that processing syntax method forms later are
+    // better-typed. It would be preferable to do this in naming, but the typing machinery makes it
+    // much easier to enforce the typeclass-like constraints.
+    for (key, mdef) in modules.key_cloned_iter_mut() {
+        validate_syntax_methods(context, &key, mdef)
+    }
```

Comment:
Do we need to restrict the ProgramInfo in some way after doing this?

Reply:
How so?

Reply:
My question is do we want to remove the invalid ones from the programinfo